app/goal_routes.py

```diff
@@ -0,0 +1,93 @@
+from app import db
+from app.models.task import Task
+from flask import Blueprint, request, make_response, abort, jsonify
+from app.routes_helpers import validate_model
+from app.models.goal import Goal
+
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if not request_body.get("title"):
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    goal = Goal.from_dict(request_body)
+
+    db.session.add(goal)
+    db.session.commit()
+
+    return make_response(goal.to_dict(), 201)
+
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+
+    goals_response = [goal.to_dict()["goal"] for goal in goals]
+
+    return jsonify(goals_response), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    return goal.to_dict(), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    request_body = request.get_json()
+ 
+    goal = validate_model(Goal, goal_id)
+ 
+    if request_body["title"]:
+        goal.title = request_body["title"]
```

Comment:
Let's flip the logic and make this more of a standard guard clause:
```python
    if not request_body["title"]:
        abort(make_response({"details": "Invalid data"}, 400))
    
    goal.title = request_body["title"]
```
Like this, we bring the main focus of the function outside the indentation.